src/java/org/apache/cassandra/transport/Server.java

```diff
@@ -404,6 +421,28 @@ public void release()
             if (-1 == refCount.updateAndGet(i -> i == 1 ? -1 : i - 1))
                 requestPayloadInFlightPerEndpoint.remove(endpoint, this);
         }
+
+        /**
+         * This will recompute the ip usage histo on each query of the snapshot when requested instead of trying to keep
+         * a histogram up to date with each request
+         */
+        public static class IpUsageReservoir implements Reservoir
```

Comment:
feel that this would be best as a private class in `ClientMetrics` as its not reusable and only to mock out the histogram state.